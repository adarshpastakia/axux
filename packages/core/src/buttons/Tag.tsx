// @author    : Adarsh Pastakia
// @version   : 0.0.1
// @copyright : 2021
// @license   : MIT

import { isColor } from "@axux/utilities";
import { FC, forwardRef, MouseEventHandler, UIEventHandler, useMemo } from "react";
import { AxIcon } from "../icons/Icon";
import { BadgeType, useBadge } from "../internals/useBadge";
import { useWithTooltip, WithTooltipProps } from "../internals/useWithTooltip";
import { BaseColor, Color, ColorPalette, ElementProps, IconProps, RefProp, Size } from "../types";
import { AppIcons } from "../types/appIcons";

/** @internal */
export interface TagProps extends RefProp, IconProps, ElementProps, WithTooltipProps {
  /**
   * Tag label color
   */
  color?: Color | BaseColor | ColorPalette | string;
  /**
   * Size
   */
  size?: Size;
  /**
   * Badge indicator
   */
  badge?: BadgeType;
  /**
   * Disable click
   */
  isDisabled?: boolean;
  /**
   * Fill color for background
   */
  fillColor?: boolean;
  /**
   * Click event handler
   */
  onClick?: MouseEventHandler;
  /**
   * Remove event handler
   */
  onRemove?: UIEventHandler;
  /**
   * Tab index to allow focus
   */
  tabIndex?: number;
}

/**
 * Label tags
 * @internal
 */
export const AxTag: FC<TagProps> = forwardRef<HTMLElement, TagProps>(
  (
    {
      icon,
      children,
      isDisabled,
      onClick,
      onRemove,
      badge,
      fillColor = false,
      size = "normal",
      color = "default",
      className,
      tooltip,
      ...aria
    },
    ref
  ) => {
    const { Wrapper, tooltipProps } = useWithTooltip(tooltip, ref);
    const badgeEl = useBadge(badge);
    const classes = useMemo(() => {
      const cls = ["ax-tag", className];
      if (color && !isColor(color)) {
        cls.push(`ax-color--${color}`);
      } else {
        cls.push("ax-color--dark");
      }
      return cls.join(" ");
    }, [className, color]);
    const styles = useMemo(() => {
      const s: KeyValue = {};
      if (color && isColor(color)) {
        s.color = color;
      }
      return s;
    }, [color]);

    return (
      <Wrapper {...tooltipProps}>
        <div
          ref={ref as AnyObject}
          className={classes}
          data-size={size}
          data-clickable={!!onClick}
          data-solid={fillColor}
          data-disabled={isDisabled}
          style={styles}
          {...aria}
        >
          <div className="ax-tag__inner" onClick={onClick}>
            {icon && <AxIcon icon={icon} />}
            <span>{children}</span>
          </div>
          {badgeEl}
          {onRemove && (
            <AxIcon
              role="remove"
              className="ax-tag__close"
              icon={AppIcons.iconClose}
              onClick={onRemove}
            />
          )}
        </div>
      </Wrapper>
    );
  }
);
AxTag.displayName = "AxTag";
