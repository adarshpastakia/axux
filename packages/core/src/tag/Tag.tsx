/**
 * AxUX React UI Framework with Tailwind CSS
 * @author    : Adarsh Pastakia
 * @version   : 4.0.0
 * @copyright : 2024
 * @license   : MIT
 */

import { isColor } from "@axux/utilities";
import { handleClick, handleEnter } from "@axux/utilities/src/handlers";
import { forwardRef, useMemo, type FC } from "react";
import { getTooltipProps } from "../hooks/useTooltip";
import { AxIcon } from "../icons/Icon";
import { Close } from "../internal/Close";
import {
  type ChildrenProp,
  type Color,
  type ElementProps,
  type EmptyCallback,
  type IconProp,
  type MouseProps,
  type RefProp,
  type TooltipType,
} from "../types";

export interface TagProps
  extends RefProp,
    ElementProps,
    MouseProps,
    IconProp,
    ChildrenProp {
  /**
   * remove handler
   */
  onRemove?: EmptyCallback;
  /**
   * disable clickable tag
   */
  isDisabled?: boolean;
  /**
   * tooltip
   */
  tooltip?: TooltipType;
  /**
   * size
   */
  size?: "sm" | "md" | "normal";
  /**
   * fill color
   */
  fill?: boolean;
  /**
   * color
   */
  color?: Color | string;
}

/**
 * Tag labels
 */
export const AxTag: FC<TagProps> = forwardRef<HTMLDivElement, TagProps>(
  (
    {
      children,
      className,
      fill,
      size,
      icon,
      rtlFlip,
      color,
      tooltip,
      onClick,
      onRemove,
      isDisabled,
      stopPropagation,
      // @ts-expect-error ignore
      "data-popover-open": popoverOpen,
      ...rest
    },
    ref,
  ) => {
    /** ***************** build style map *******************/
    const styles = useMemo(() => {
      const s: KeyValue = {};
      if (color && isColor(color)) {
        s["--color"] = color;
      }
      return s;
    }, [color]);

    const tooltipProps = useMemo(() => getTooltipProps(tooltip), [tooltip]);

    const clickHandler = useMemo(() => {
      if (!isDisabled && onClick) {
        return handleClick(onClick, { stopPropagation });
      }
    }, [onClick, isDisabled, stopPropagation]);

    /** ***************** component *******************/
    return (
      <div
        {...rest}
        {...tooltipProps}
        ref={ref}
        role={onClick == null ? "term" : "button"}
        tabIndex={-1}
        className={`ax-tag ${className ?? ""}`}
        data-fill={fill}
        data-size={size}
        data-color={color}
        data-clickable={!(onClick == null)}
        data-disabled={isDisabled}
        data-popover-open={popoverOpen}
        onClick={clickHandler}
        onKeyDown={handleEnter(onClick)}
        style={styles}
      >
        {icon && <AxIcon icon={icon} rtlFlip={rtlFlip} />}
        <label>{children}</label>
        {onRemove != null && Close(onRemove)}
      </div>
    );
  },
) as AnyObject;
AxTag.displayName = "AxTag";
