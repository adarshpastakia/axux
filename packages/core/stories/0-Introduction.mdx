import { Title, Canvas, Story } from "@storybook/addon-docs";
import { Anchor, Subheading } from "@storybook/blocks";
import { Meta } from "@storybook/blocks";

<Meta title="@core/Introduction" />

# Core Components

<Anchor storyId="install">
  <Subheading>Installing</Subheading>
</Anchor>

```bash

#npm
npm install @axux/core

# yarn
yarn add @axux/core

```

<Anchor storyId="using">
  <Subheading>Using AxUX</Subheading>
</Anchor>

```tsx
import { AxApplicationProvider,AxViewport } from "@axux/core";
import { I18nextProvider } from "react-i18next";
import { BrowserRouter } from "react-router";

export const Application = () => {
  return (
    {/** i18n for localization **/}
    <I18nextProvider>
      {/** react router **/}
      <BrowserRouter>
        <AxApplicationProvider>
          <AxViewport>
            .
            .
            .
            .
          </AxViewport>
        </AxApplicationProvider>
      </BrowserRouter>
    </I18nextProvider>
  );
};
```
