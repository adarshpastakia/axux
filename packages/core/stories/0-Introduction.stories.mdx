import { Title, Canvas, Story } from "@storybook/addon-docs";

<Meta title="@core/Introduction" />

<Title>Core Components</Title>
