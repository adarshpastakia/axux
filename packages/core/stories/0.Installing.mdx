import { Title, Canvas, Story } from "@storybook/addon-docs";
import { Anchor, Subheading } from "@storybook/blocks";
import { Meta } from "@storybook/blocks";

<Meta title="@core/Installing" />

# Core Components

<Anchor storyId="install">
  <Subheading>Installing</Subheading>
</Anchor>

```bash

#npm
npm install @axux/core

# yarn
yarn add @axux/core

```

<Anchor storyId="using">
  <Subheading>Using AxUX</Subheading>
</Anchor>

<cite type="code">src/index.tsx</cite>

```tsx
import { AxApplicationProvider, AxViewport } from "@axux/core";
import { Fragment, StrictMode } from "react";
import ReactDOM from "react-dom/client";
import { I18nextProvider } from "react-i18next";
import { Outlet, RouterProvider, createMemoryRouter } from "react-router";
import i18next from "./i18n";
import reportWebVitals from "./reportWebVitals";
import "./styles.css";

const root = ReactDOM.createRoot(document.getElementById("root"));

const router = createMemoryRouter([
  {
    path: "/",
    element: (
      <AxViewport>
        <div>Build the ui here</div>
        <Outlet />
      </AxViewport>
    ),
    children: [
      {
        path: "page",
        element: <Page />,
      },
      {
        path: "page",
        element: <Page />,
      },
    ],
  },
]);

const Wrapper = process.env.NODE_ENV === "development" ? Fragment : StrictMode;
root.render(
  <Wrapper>
    <AxApplicationProvider defaultPrimary="denim" defaultAccent="coral">
      <I18nextProvider i18n={i18next}>
        <RouterProvider router={router} />
      </I18nextProvider>
    </AxApplicationProvider>
  </Wrapper>,
);

// @ts-expect-error ignore
if (module.hot) {
  // @ts-expect-error ignore
  module.hot.accept();
}

// If you want to start measuring performance in your app, pass a function
// to log results (for example: reportWebVitals(console.log))
// or send to an analytics endpoint. Learn more: https://bit.ly/CRA-vitals
reportWebVitals(console.debug);
```
