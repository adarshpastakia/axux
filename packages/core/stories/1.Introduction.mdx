import {
  Title,
  Canvas,
  Story,
  Description,
  Heading,
  Subheading,
  Anchor,
} from "@storybook/addon-docs";
import { Meta } from "@storybook/blocks";
import {
  AxAnimation,
  AxApplicationProvider,
  AxAside,
  AxBreadcrumb,
  AxButton,
  AxCallout,
  AxCard,
  AxCollapsable,
  AxContent,
  AxDivider,
  AxErrorBoundary,
  AxFooter,
  AxHeader,
  AxFlexBox,
  AxLoader,
  AxIcon,
  AxMenu,
  AxMeter,
  AxSection,
  AxPage,
  AxProgress,
  AxTag,
  AxText,
  AxTheme,
  AxTitle,
  AxViewport,
  useApplicationContext,
} from "../src";

<Meta title="@core/Introduction" />

# Core Components

<Anchor storyId="app">
  <Subheading>Application</Subheading>
</Anchor>

<div className="card-grid">
  <div className="comp-card">
    <label>AxApplicationProvider</label>
    <Description of={AxApplicationProvider} />
  </div>
  <div className="comp-card">
    <label>AxViewport</label>
    <Description of={AxViewport} />
  </div>
  <div className="comp-card">
    <label>AxPage</label>
    <Description of={AxPage} />
  </div>
  <div className="comp-card">
    <label>AxErrorBoundary</label>
    <Description of={AxErrorBoundary} />
  </div>
</div>

<Anchor storyId="pageLayouts">
  <Subheading>Page Layout</Subheading>
</Anchor>

<div className="card-grid">
  <div className="comp-card">
    <label>AxSection</label>
    <Description of={AxSection} />
  </div>
  <div className="comp-card">
    <label>AxContent</label>
    <Description of={AxContent} />
  </div>
  <div className="comp-card">
    <label>AxAside</label>
    <Description of={AxAside} />
  </div>
  <div className="comp-card">
    <label>AxHeader</label>
    <Description of={AxHeader} />
  </div>
  <div className="comp-card">
    <label>AxFooter</label>
    <Description of={AxFooter} />
  </div>
</div>

<Anchor storyId="flexLayouts">
  <Subheading>Flex Layout</Subheading>
</Anchor>

<div className="card-grid">
  <div className="comp-card">
    <label>AxFlexBox</label>
    <Description of={AxFlexBox} />
  </div>
  <div className="comp-card">
    <label>AxFlexBox.Row</label>
    <Description of={AxFlexBox.Row} />
  </div>
  <div className="comp-card">
    <label>AxFlexBox.Col</label>
    <Description of={AxFlexBox.Col} />
  </div>
</div>

<Anchor storyId="typography">
  <Subheading>Typography</Subheading>
</Anchor>

<div className="card-grid">
  <div className="comp-card">
    <label>AxText</label>
    <Description of={AxText} />
  </div>
  <div className="comp-card">
    <label>AxText.Abbr</label>
    <Description of={AxText.Abbr} />
  </div>
  <div className="comp-card">
    <label>AxText.Mark</label>
    <Description of={AxText.Mark} />
  </div>
  <div className="comp-card">
    <label>AxText.Copy</label>
    <Description of={AxText.Copy} />
  </div>
  <div className="comp-card">
    <label>AxTitle</label>
    <Description of={AxTitle} />
  </div>
</div>

<Anchor storyId="buttons">
  <Subheading>Buttons</Subheading>
</Anchor>

<div className="card-grid">
  <div className="comp-card">
    <label>AxButton</label>
    <Description of={AxButton} />
  </div>
  <div className="comp-card">
    <label>AxButton.Group</label>
    <Description of={AxButton.Group} />
  </div>
  <div className="comp-card">
    <label>AxButton.Confirm</label>
    <Description of={AxButton.Confirm} />
  </div>
  <div className="comp-card">
    <label>AxButton.Action</label>
    <Description of={AxButton.Action} />
  </div>
  <div className="comp-card">
    <label>AxButton.Dropdown</label>
    <Description of={AxButton.Dropdown} />
  </div>
</div>
