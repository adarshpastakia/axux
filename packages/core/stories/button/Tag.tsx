/**
 * AxUX React+TailwindCSS UI Framework
 * @author    : Adarsh Pastakia
 * @version   : 2.0.0
 * @copyright : 2022
 * @license   : MIT
 */

import { ComponentStory } from "@storybook/react";
import { AxTag } from "../../src";

export const TagTemplate: ComponentStory<typeof AxTag> = (props) => (
  <AxTag {...props} />
);

export const TagStory = TagTemplate.bind({});
TagStory.args = {
  onClick: undefined,
  onRemove: undefined,
  children: "Tag Label",
};
export const ColorStory = TagTemplate.bind({});
ColorStory.args = {
  icon: "mdi mdi-bell",
  children: "Tag Label",
  color: "primary",
};
export const ColorValueStory = TagTemplate.bind({});
ColorValueStory.args = {
  icon: "mdi mdi-bell",
  children: "Tag Label",
  color: "#1d4ed8",
};

export default { title: "AxTag", component: AxTag };
