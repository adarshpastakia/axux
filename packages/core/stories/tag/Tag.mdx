import {
  ArgsTable,
  Canvas,
  Description,
  Primary,
  Story,
  Title,
} from "@storybook/addon-docs";
import {
  Anchor,
  ArgTypes,
  Controls,
  Meta,
  Subheading,
} from "@storybook/blocks";
import { AxTag, AxCollapsable } from "../../src";
import * as TagStories from "./Tag.stories";

<Meta of={TagStories} />

<Anchor storyId="tag">
  <div className="invisible h-0 overflow-hidden">
    <Subheading>Tag</Subheading>
  </div>
</Anchor>

# Tag Labels

<Description of={AxTag} />

<Primary />

<AxCollapsable>
  <div>Controls</div>
  <Controls of={TagStories.Example} />
</AxCollapsable>

<Anchor storyId="variants">
  <Subheading>Variants</Subheading>
</Anchor>
<Canvas withToolbar of={TagStories.Variants} />

<Anchor storyId="colors">
  <Subheading>Colors</Subheading>
</Anchor>
<Canvas withToolbar of={TagStories.Colors} />

<AxCollapsable>
  <div>Controls</div>
  <Controls of={TagStories.Colors} />
</AxCollapsable>

<Anchor storyId="sizes">
  <Subheading>Sizes</Subheading>
</Anchor>
<Canvas withToolbar of={TagStories.Sizes} />

<Anchor storyId="states">
  <Subheading>States</Subheading>
</Anchor>
<Canvas withToolbar of={TagStories.States} />

<hr />

<Anchor storyId="props">
  <Subheading>Props</Subheading>
</Anchor>

<ArgTypes sort="requiredFirst" of={AxTag} />

<Anchor storyId="sample">
  <Subheading>Example</Subheading>
</Anchor>

```tsx
import { AxTag } from "@axux/core";

export const TagExample = () => {
  return (
    <AxTag
      color="primary"
      fill={fillState}
      icon={svgPath || "mdi mdi-icon" || "image.url"}
      isDisabled={disabledState}
      size="md"
      tooltip={{ content: "Tooltip", color: "info", placement: "top" }}
      onClick={clickHandler}
      onRemove={removeHandler}
      stopPropagation
    >
      Tag Label
    </AxTag>
  );
};
```

import { Children } from "react";
