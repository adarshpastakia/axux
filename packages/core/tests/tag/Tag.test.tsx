/**
 * AxUX React UI Framework with Tailwind CSS
 * @author    : Adarsh Pastakia
 * @version   : 4.0.0
 * @copyright : 2024
 * @license   : MIT
 */

import { composeStories } from "@storybook/testing-react";
import { fireEvent, render } from "@testing-library/react";
import React from "react";
import * as stories from "../../stories/tag/Tag.stories";

const { Example } = composeStories(stories);

describe("Tag", () => {
  it("should render", () => {
    const fragment = render(<Example>Simple Tag</Example>);
    expect(fragment.container.querySelector(".ax-tag")).not.toBeNull();
    expect(fragment.container.querySelector(".ax-tag > .ax-icon")).toBeNull();
    expect(fragment.container.querySelector(".ax-tag > label")?.innerHTML).toBe(
      "Simple Tag",
    );
    expect(document.body.innerHTML).toMatchSnapshot();
    fragment.unmount();
  });

  it("should render with icon", () => {
    const fragment = render(<Example icon="mdi mdi-bell" />);
    expect(
      fragment.container.querySelector(".ax-tag > .ax-icon"),
    ).not.toBeNull();
    expect(document.body.innerHTML).toMatchSnapshot();
    fragment.unmount();
  });

  it("should render icon and label", () => {
    const fragment = render(<Example icon="mdi mdi-bell">Click Me</Example>);
    expect(
      fragment.container.querySelector(".ax-tag > .ax-icon"),
    ).not.toBeNull();
    expect(fragment.container.querySelector(".ax-tag > label")).not.toBeNull();
    expect(document.body.innerHTML).toMatchSnapshot();
    fragment.unmount();
  });

  it("should be clickable", () => {
    const fn = jest.fn();
    const fragment = render(<Example onClick={fn}>Click Me</Example>);
    fireEvent.click(fragment.container.querySelector(".ax-tag") as HTMLElement);
    expect(fn).toHaveBeenCalled();
    expect(document.body.innerHTML).toMatchSnapshot();
    fragment.unmount();
  });

  it("should be closable", () => {
    const fn = jest.fn();
    const fragment = render(<Example onRemove={fn}>Click Me</Example>);
    expect(
      fragment.container.querySelector(".ax-tag > .close-x"),
    ).not.toBeNull();
    fireEvent.click(
      fragment.container.querySelector(".ax-tag > .close-x") as HTMLElement,
    );
    expect(fn).toHaveBeenCalled();
    expect(document.body.innerHTML).toMatchSnapshot();
    fragment.unmount();
  });

  it("should be disabled", () => {
    const fn = jest.fn();
    const fragment = render(
      <Example isDisabled onClick={fn}>
        Click Me
      </Example>,
    );
    expect(
      fragment.container.querySelector('.ax-tag[data-disabled="true"]'),
    ).not.toBeNull();
    fireEvent.click(fragment.container.querySelector(".ax-tag") as HTMLElement);
    expect(fn).not.toHaveBeenCalled();
    expect(document.body.innerHTML).toMatchSnapshot();
    fragment.unmount();
  });
});
