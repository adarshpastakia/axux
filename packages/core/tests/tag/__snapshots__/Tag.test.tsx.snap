// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag should be clickable 1`] = `"<div></div><div></div><div></div><div><div class="contents" data-color-scheme="light" data-primary-scheme="denim" data-accent-scheme="jade"><div role="button" tabindex="-1" class="ax-tag " data-clickable="true"><label>Click Me</label><span class="close-x" role="button" aria-label="close" tabindex="-1"><span role="presentation" class="ax-icon "><svg viewBox="0 0 24 24"><path fill="currentColor" d="M5.293 6.707l5.293 5.293-5.293 5.293c-0.391 0.391-0.391 1.024 0 1.414s1.024 0.391 1.414 0l5.293-5.293 5.293 5.293c0.391 0.391 1.024 0.391 1.414 0s0.391-1.024 0-1.414l-5.293-5.293 5.293-5.293c0.391-0.391 0.391-1.024 0-1.414s-1.024-0.391-1.414 0l-5.293 5.293-5.293-5.293c-0.391-0.391-1.024-0.391-1.414 0s-0.391 1.024 0 1.414z"></path></svg></span></span></div><div class="ax-overlay__container" data-mode="overlay"></div><div class="ax-overlay__container" data-mode="alert"></div><div class="ax-notification__container" data-mode="toast"></div><div class="ax-notification__container" data-mode="message"></div></div></div>"`;

exports[`Tag should be closable 1`] = `"<div></div><div></div><div></div><div></div><div><div class="contents" data-color-scheme="light" data-primary-scheme="denim" data-accent-scheme="jade"><div role="button" tabindex="-1" class="ax-tag " data-clickable="true"><label>Click Me</label><span class="close-x" role="button" aria-label="close" tabindex="-1"><span role="presentation" class="ax-icon "><svg viewBox="0 0 24 24"><path fill="currentColor" d="M5.293 6.707l5.293 5.293-5.293 5.293c-0.391 0.391-0.391 1.024 0 1.414s1.024 0.391 1.414 0l5.293-5.293 5.293 5.293c0.391 0.391 1.024 0.391 1.414 0s0.391-1.024 0-1.414l-5.293-5.293 5.293-5.293c0.391-0.391 0.391-1.024 0-1.414s-1.024-0.391-1.414 0l-5.293 5.293-5.293-5.293c-0.391-0.391-1.024-0.391-1.414 0s-0.391 1.024 0 1.414z"></path></svg></span></span></div><div class="ax-overlay__container" data-mode="overlay"></div><div class="ax-overlay__container" data-mode="alert"></div><div class="ax-notification__container" data-mode="toast"></div><div class="ax-notification__container" data-mode="message"></div></div></div>"`;

exports[`Tag should be disabled 1`] = `"<div></div><div></div><div></div><div></div><div></div><div><div class="contents" data-color-scheme="light" data-primary-scheme="denim" data-accent-scheme="jade"><div role="button" tabindex="-1" class="ax-tag " data-clickable="true" data-disabled="true"><label>Click Me</label><span class="close-x" role="button" aria-label="close" tabindex="-1"><span role="presentation" class="ax-icon "><svg viewBox="0 0 24 24"><path fill="currentColor" d="M5.293 6.707l5.293 5.293-5.293 5.293c-0.391 0.391-0.391 1.024 0 1.414s1.024 0.391 1.414 0l5.293-5.293 5.293 5.293c0.391 0.391 1.024 0.391 1.414 0s0.391-1.024 0-1.414l-5.293-5.293 5.293-5.293c0.391-0.391 0.391-1.024 0-1.414s-1.024-0.391-1.414 0l-5.293 5.293-5.293-5.293c-0.391-0.391-1.024-0.391-1.414 0s-0.391 1.024 0 1.414z"></path></svg></span></span></div><div class="ax-overlay__container" data-mode="overlay"></div><div class="ax-overlay__container" data-mode="alert"></div><div class="ax-notification__container" data-mode="toast"></div><div class="ax-notification__container" data-mode="message"></div></div></div>"`;

exports[`Tag should render 1`] = `"<div><div class="contents" data-color-scheme="light" data-primary-scheme="denim" data-accent-scheme="jade"><div role="button" tabindex="-1" class="ax-tag " data-clickable="true"><label>Simple Tag</label><span class="close-x" role="button" aria-label="close" tabindex="-1"><span role="presentation" class="ax-icon "><svg viewBox="0 0 24 24"><path fill="currentColor" d="M5.293 6.707l5.293 5.293-5.293 5.293c-0.391 0.391-0.391 1.024 0 1.414s1.024 0.391 1.414 0l5.293-5.293 5.293 5.293c0.391 0.391 1.024 0.391 1.414 0s0.391-1.024 0-1.414l-5.293-5.293 5.293-5.293c0.391-0.391 0.391-1.024 0-1.414s-1.024-0.391-1.414 0l-5.293 5.293-5.293-5.293c-0.391-0.391-1.024-0.391-1.414 0s-0.391 1.024 0 1.414z"></path></svg></span></span></div><div class="ax-overlay__container" data-mode="overlay"></div><div class="ax-overlay__container" data-mode="alert"></div><div class="ax-notification__container" data-mode="toast"></div><div class="ax-notification__container" data-mode="message"></div></div></div>"`;

exports[`Tag should render icon and label 1`] = `"<div></div><div></div><div><div class="contents" data-color-scheme="light" data-primary-scheme="denim" data-accent-scheme="jade"><div role="button" tabindex="-1" class="ax-tag " data-clickable="true"><span role="presentation" class="ax-icon "><i class="mdi mdi-bell"></i></span><label>Click Me</label><span class="close-x" role="button" aria-label="close" tabindex="-1"><span role="presentation" class="ax-icon "><svg viewBox="0 0 24 24"><path fill="currentColor" d="M5.293 6.707l5.293 5.293-5.293 5.293c-0.391 0.391-0.391 1.024 0 1.414s1.024 0.391 1.414 0l5.293-5.293 5.293 5.293c0.391 0.391 1.024 0.391 1.414 0s0.391-1.024 0-1.414l-5.293-5.293 5.293-5.293c0.391-0.391 0.391-1.024 0-1.414s-1.024-0.391-1.414 0l-5.293 5.293-5.293-5.293c-0.391-0.391-1.024-0.391-1.414 0s-0.391 1.024 0 1.414z"></path></svg></span></span></div><div class="ax-overlay__container" data-mode="overlay"></div><div class="ax-overlay__container" data-mode="alert"></div><div class="ax-notification__container" data-mode="toast"></div><div class="ax-notification__container" data-mode="message"></div></div></div>"`;

exports[`Tag should render with icon 1`] = `"<div></div><div><div class="contents" data-color-scheme="light" data-primary-scheme="denim" data-accent-scheme="jade"><div role="button" tabindex="-1" class="ax-tag " data-clickable="true"><span role="presentation" class="ax-icon "><i class="mdi mdi-bell"></i></span><label>Tag Label</label><span class="close-x" role="button" aria-label="close" tabindex="-1"><span role="presentation" class="ax-icon "><svg viewBox="0 0 24 24"><path fill="currentColor" d="M5.293 6.707l5.293 5.293-5.293 5.293c-0.391 0.391-0.391 1.024 0 1.414s1.024 0.391 1.414 0l5.293-5.293 5.293 5.293c0.391 0.391 1.024 0.391 1.414 0s0.391-1.024 0-1.414l-5.293-5.293 5.293-5.293c0.391-0.391 0.391-1.024 0-1.414s-1.024-0.391-1.414 0l-5.293 5.293-5.293-5.293c-0.391-0.391-1.024-0.391-1.414 0s-0.391 1.024 0 1.414z"></path></svg></span></span></div><div class="ax-overlay__container" data-mode="overlay"></div><div class="ax-overlay__container" data-mode="alert"></div><div class="ax-notification__container" data-mode="toast"></div><div class="ax-notification__container" data-mode="message"></div></div></div>"`;
